tests/unit/FirebaseStorage.spec.js

```diff
@@ -0,0 +1,284 @@
+'use strict';
+describe('$firebaseStorage', function () {
+  var $firebaseStorage;
+  var URL = 'https://oss-test.firebaseio.com';
+
+  beforeEach(function () {
+    module('firebase.storage');
+  });
+
+  describe('<constructor>', function () {
+
+    var $firebaseStorage;
+    var $q;
+    var $rootScope;
+    var $firebaseUtils;
+    beforeEach(function () {
+      module('firebase.storage');
+      inject(function (_$firebaseStorage_, _$q_, _$rootScope_, _$firebaseUtils_) {
+        $firebaseStorage = _$firebaseStorage_;
+        $q = _$q_;
+        $rootScope = _$rootScope_;
+        $firebaseUtils = _$firebaseUtils_;
+      });
+    });
+
+    it('should exist', inject(function () {
+      expect($firebaseStorage).not.toBe(null);
+    }));
+
+    it('should create an instance', function () {
+      var ref = firebase.storage().ref('thing');
+      var storage = $firebaseStorage(ref);
+      expect(storage).not.toBe(null);
+    });
+
+    describe('$firebaseStorage.utils', function () {
+
+      describe('_unwrapStorageSnapshot', function () {
+
+        it('should unwrap the snapshot', function () {
```

Comment:
We should probably add some extra attributes into the source snapshot to make sure the method actually removes them.